### ISSUE TITLE
Make Try reactive

### ISSUE THREAD

**Issue by johnmcclean-aol** [aol/cyclops-react#638]
Try can be made optionally

* eager
* lazy
* reactive

by leveraging Xor as the underlying data type and making use of Either to represent lazy or reactive Try instances.

**Comment by johnmcclean-aol** [comment]
Example to load data from an external service that may fail, and catch only IOExceptions
```java
public Try<Integer, IOException> serviceCall(){ 
      return Future.ofSupplier(this::loadData,executor).toTry(IOException.class);
}
```

**Comment by johnmcclean-aol** [comment]
Released in 2.0.0-FINAL
